src/test/scala/mesosphere/marathon/integration/EventsIntegrationTest.scala

```diff
@@ -0,0 +1,63 @@
+package mesosphere.marathon
+package integration
+
+import java.util.UUID
+import java.util.concurrent.ConcurrentLinkedQueue
+
+import mesosphere.AkkaIntegrationTest
+import mesosphere.marathon.integration.facades.{ ITConnected, ITEvent, ITSSEEvent }
+import mesosphere.marathon.integration.setup._
+import mesosphere.marathon.state.PathId
+import mesosphere.marathon.state.PathId._
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+@IntegrationTest
+class EventsIntegrationTest extends AkkaIntegrationTest with EmbeddedMarathonTest {
+
+  before(cleanUp())
+
+  def appId(): PathId = testBasePath / s"app-${UUID.randomUUID}"
+
+  "Filter events" should {
+    "receive only app deployment event" in {
+      Given("a duration limit for connect")
+      val atMost = patienceConfig.timeout.toMillis.millis
+
+      Given("a new event source without filter")
+      val source = Await.result(marathon.events(), atMost)
+
+      Given("a new event source with filter")
+      val filteredSource = Await.result(marathon.events("deployment_success"), atMost)
+
+      When("Gather subscribed events")
+
+      def enqueue(e: ITSSEEvent, queue: ConcurrentLinkedQueue[String]) = {
+        e match {
+          case ITConnected => ???
+          case event: ITEvent => queue.offer(event.eventType)
+        }
+      }
+
+      val filteredEventsQueue = new ConcurrentLinkedQueue[String]()
+      filteredSource.runForeach(enqueue(_, filteredEventsQueue))
+      val eventsQueue = new ConcurrentLinkedQueue[String]()
+      source.runForeach(enqueue(_, eventsQueue))
+
+      Given("a new app")
+      val app = appProxy(appId(), "v1", instances = 1, healthCheck = None)
+
+      When("The app is deployed")
+      val result = marathon.createAppV2(app)
+
+      Then("Wait for deployment")
+      waitForDeployment(result)
+      waitForTasks(app.id.toPath, 1)
```

Comment:
Any idea how to fix it? Events should be published at once so after we wait for deployment to finish and tasks messages should be delivered. There is a chance events will be queued and delivered after but IMO this will be a bug.